Fix queryId calculation for queries with grouping clause

whitehawk opened the pull request:
Problem description:
if 'pg_stat_statements' extension is enabled and a query containing GROUP BY
clause with ROLLUP, CUBE or GROUPING SETS or with GROUPING function is
executed, queryId calculation doesn't take into account these grouping clause
parameters and GROUPING functions. So, semantically different queries are
treated by pg_stat_statements as equivalent queries. Plus warning messages
appear during jumbling.
Also, jumbling logic doesn't handle properly queries with group_id() function
call and calls of functions with 'anytable' parameter.
First type of queries causes warnings.
Second type of query causes error message "unrecognized RTE kind: 7".

Expected correct behavior: queries with different grouping clauses or different
usage of the GROUPING function are treated by pg_stat_statements as different
queries, and no warning messages appear during the query execution.

Cause:
In JumbleExpr there is no handling for node tags T_GroupingClause,
T_GroupingFunc, T_GroupId and T_TableValueExpr. Jumble hashing is used
as queryId. Error message "unrecognized RTE kind: 7" caused by not handled 
range table type RTE_TABLEFUNCTION in JumbleRangeTable.

Fix:
Handling logic for missed node tags was added.
According to comments in queryjumble.c, main guideline how to handle
query tree is: "Rule of thumb for what to include is that we should ignore
anything not semantically significant (such as alias names) as well as
anything that can be deduced from child nodes (else we'd just be double-
hashing that piece of information)."

For T_GroupingFunc we append to the jumble the list of
arguments. Field 'ngrpcols' is not appended to the jumble, because it can be
deduced from the list of groupsets in grouping clause. 'ngrpcols' is the
number of unique grouping attributes in grouping clause. Equivalent queries
must have the same groupsets in grouping clause. Thus they will have the same
'ngrpcols'. So, adding 'ngrpcols' to jumble is redundant.
Plus handling for the T_Integer tag was added, because it is required to parse
the list of arguments. List of arguments (field 'args' of structure
GroupingFunc) is a list of T_Integer. The T_Integer element is the index of
GROUPING function parameter inside the array of unique grouping attributes
from grouping clause. We need to jumble value of this T_Integer, because
changing parameter in GROUPING function changes this index (and definitely
changes semantic of the query).

For T_GroupingClause we append to the jumble the grouping type
and list of groupsets. Field 'location' is not appended to the jumble because
it is the textual location from parser and is not semantically significant.

For T_GroupId - this tag was added to switch case inside JumbleExpr to
suppress warning. The node tag for it was already handled. Struct GroupId
doesn't have any additional fields, that could be added to jumble.

For RTE_TABLEFUNCTION - 'functions' field of RangeTblEntry was added to the
jumble. RangeTblEntry in case of RTE_TABLEFUNCTION uses 'functions' and
'subquery' fields (other significant fields are null). But 'subquery' is
duplicated in TableValueExpr below, so do not jumble it.

For T_TableValueExpr - 'subquery' field of TableValueExpr (the subquery
that is inside "TABLE(<subquery>)" statement) was added to the jumble.

whitehawk committed:
Fix warnings in Jumbling logic

Problem description:
Following warning messages appear
    "WARNING: unrecognized node type: 925"
    "WARNING: unrecognized node type: 924"
if 'pg_stat_statements' extension is enabled and query,
containing GROUP BY clause or GROUPING function, is executed.

Cause:
In JumbleExpr, called from JumbleQuery, there is no handling
for node tags T_GroupingClause and T_GroupingFunc.

Fix:
Added handling logic for missed node tags.
    For T_GroupingFunc we append to the jumble the tag itself
and list of arguments. Field 'ngrpcols' is not appended to the
jumble, because it can be deduced from the list of arguments
in grouping clause. Plus added handling for T_Integer tag,
because it is required to parse list of arguments.
    For T_GroupingClause we append to the jumble the tag itself,
grouping type and list of groupsets. Field 'location' is not
appended to the jumble, because it is the textual location from
parser and is not semantically significant.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/60788

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/909714

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/909715

whitehawk committed:
Add simple test for pg_stat_statements

If enabled, pg_stat_statements may cause warnings in case jumbling logic
doesn't handle some part of a query. This simple test will check that
no warnings are generated on different queries. Currently added 1 query
with GROUP clause and GROUPING function, as it recently caused such issues.
This test later may be improved by adding more test queries.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/60873

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/915047

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/915048

whitehawk committed:
Fix inconsistent register in the query of the pg_stat_statements test

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/60887

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/915839

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/917776

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/920047

andr-sokolov commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,21 @@
+-- simple test to check there is no warnings during jumbling of the query
+CREATE TABLE table_test_pg_stat_statements
```
Queries in tests should be as simple as possible, but sufficient to detect an error.
We should drop table before creating it, because the table may exist. For example
```
-- start_ignore
DROP TABLE IF EXISTS t;
-- end_ignore

CREATE TABLE t (a int) DISTRIBUTED BY (a);
SELECT GROUPING(a) FROM t GROUP BY GROUPING SETS ((a));
DROP TABLE t;
```

now the test is passed before the changes in queryjumble.c

whitehawk replied:
test was updated to cover this comment

whitehawk committed:
Improve test for pg_stat_statements

Modified the pg_stat_statements test to enable the pg_stat_statements
extension automatically at the beginning of the test, and disable the
extension once the test is complete.
Also added minor refactoring of recently added code in JumbleExpr - moved
all new code to the end of the switch case in order to simplify the patch.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61084

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/928426

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/928427

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/928417

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/928525

KnightMurloc commented on src/backend/utils/misc/queryjumble.c:
```diff
@@ -597,6 +597,24 @@ JumbleExpr(JumbleState *jstate, Node *node)
 				JumbleExpr(jstate, rtfunc->funcexpr);
 			}
 			break;
+		case T_Integer:
+			{
+				APP_JUMB(intVal(node));
+			}
+			break;
```
Add to the description why we need to handle this type of node and why it's implemented that way.

whitehawk replied:
Updated description with the requested information

KnightMurloc commented:
Can you provide a test not only to check for the absence of warnings, but also that the query text in pg_stat_statements differs with and without a patch (if possible). To check the correctness of the logic implementation.

RekGRpth commented on src/backend/utils/misc/queryjumble.c:
```diff
@@ -597,6 +597,24 @@ JumbleExpr(JumbleState *jstate, Node *node)
+				APP_JUMB(intVal(node));
+			}
+			break;
+		case T_GroupingFunc:
```
According to the [commit](https://github.com/postgres/postgres/commit/5fd9dfa5f50e4906c35133a414ebec5b6d518493#diff-6a2775ed8b21d7748929e715791290669412ccae2e643bee5afb49c819c30049R398-R404), T_GroupingFunc for some reason was not backported by the commit f7ff7c8ecae4ce7ab3b73fd46171cdaa457b3591, and it does not look quite right and is located in the wrong place.

whitehawk replied:
Reason why logic for T_GroupingFunc wasn't backported in not 100% clear.
But I assume the reason is following:

ADB-6 definition of GroupingFunc:
```
typedef struct GroupingFunc
{
	NodeTag   type;
	List     *args;  /* arguments provided in the query. */
	int       ngrpcols; /* the number of grouping attributes */
} GroupingFunc;
```
is different comparing with the one from Postgres and latest GPDB:
```
typedef struct GroupingFunc
{
	Expr		xpr;
	List	   *args;			/* arguments, not evaluated but kept for
								 * benefit of EXPLAIN etc. */
	List	   *refs;			/* ressortgrouprefs of arguments */
	List	   *cols;			/* actual column positions set by planner */
	Index		agglevelsup;	/* same as Aggref.agglevelsup */
	int			location;		/* token location */
} GroupingFunc;
```
ADB-6 GroupingFunc definition is incompatible with T_GroupingFunc handling in original commit (as it refers to 'refs' field).
Thus current implementation is different as GroupingFunc is different.
The code location was discussed with Andrey - he asked to place all delta right before "default" statement.

RekGRpth commented on src/backend/utils/misc/queryjumble.c:
```diff
@@ -597,6 +597,24 @@ JumbleExpr(JumbleState *jstate, Node *node)
+		case T_GroupingFunc:
+			{
+				GroupingFunc *gf = (GroupingFunc *) node;
+				JumbleExpr(jstate, (Node *) gf->args);
```
Maybe we still need to backport the [commit](https://github.com/postgres/postgres/commit/f7a97b6ec31f3f57a6154d0039c4de81ad517064)?

whitehawk replied:
the same issue with GroupingFunc struct - in adb-6 there are no such fields that were used in mentioned commit.

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,31 @@
+-- start_ignore
+\! gpconfig -c shared_preload_libraries -v 'pg_stat_statements';
+\! gpstop -raiq;
```
the '-i' parameter is deprecated. is there any reason to use it?

whitehawk replied:
Only reason is that it was done so in sample code (tests in auto_explain). Will change to avoid using deprecated parameters.

KnightMurloc commented:
it seems to me that the problem is more that queryId is calculated incorrectly (for example, the `group by` type is not taken into account). it is worth mentioning this as a problem in the description and testing it.

whitehawk committed:
Improve test for pg_stat_statements

With this update the test for pg_stat_statements verifies that
queryId for queries with semantically different GROUP BY options are
different (thus these queries reside in seperate entries of pg_stat_statements
statistics).

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61416

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/949680

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/949681

whitehawk commented:
> Can you provide a test not only to check for the absence of warnings, but also that the query text in pg_stat_statements differs with and without a patch (if possible). To check the correctness of the logic implementation.

Test was updated.

whitehawk commented:
> it seems to me that the problem is more that queryId is calculated incorrectly (for example, the `group by` type is not taken into account). it is worth mentioning this as a problem in the description and testing it.

Description and test were updated to reflect this aspect.

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,215 @@
+)
+DISTRIBUTED BY (item1);
+
+SELECT   
+    SUM(item4) AS sum_item1
+   ,item2
+   ,GROUPING(item1)+GROUPING(item2) AS item_1_2_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
```
any reasons to leave that test when we now have the rest of the tests?

whitehawk replied:
No significant reason, can be removed

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,215 @@
+SELECT pg_stat_statements_reset();
+
+-- launch 2 equivalent queries  
+SELECT
+    GROUPING(item1) AS item_1_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
```
it seems to me that alias are not needed for this test.

whitehawk replied:
will remove alias

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,215 @@
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- change alias name
+SELECT
+    GROUPING(item2) AS item_2_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
```
changing alias will not affect queryid even without the patch. therefore, there is no point in testing it.

whitehawk replied:
will remove this part

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,215 @@
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+SET optimizer='on';
```
use `reset`

whitehawk replied:
Do you mean to use pg_stat_statements_reset() at the end of the test?

KnightMurloc replied:
i mean use `RESET optimizer;` instead of `SET optimizer='on';`

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,215 @@
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several parameters options in CUBE
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item1, item2);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item2, item3);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item3, item4);
```
the logic of attribute processing for ROLLUP and CUBE is the same, I think it's worth leaving one thing (is there anything specific in processing GROUPING SETS, if not, then they should also be removed). I also think two queries will be enough for the test, not three.

whitehawk replied:
Ok, will leave 1 option with 2 queries

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,215 @@
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several options in simple GROUP BY
```
I also don't see the point in 4 queries here.

whitehawk replied:
Ok, will reduce to 2

KnightMurloc commented on src/backend/utils/misc/queryjumble.c:
```diff
@@ -597,6 +597,24 @@ JumbleExpr(JumbleState *jstate, Node *node)
+				APP_JUMB(intVal(node));
+			}
+			break;
+		case T_GroupingFunc:
+			{
+				GroupingFunc *gf = (GroupingFunc *) node;
+				JumbleExpr(jstate, (Node *) gf->args);
+			}
+			break;
+		case T_GroupingClause:
+			{
+				GroupingClause *gc = (GroupingClause *) node;
+				APP_JUMB(gc->groupType);
+				JumbleExpr(jstate, (Node *) gc->groupsets);
+			}
+			break;
```
the order of the cases in switch is mostly ordered by enum values, so I think it makes sense to swap T_GroupingFunc and T_GroupingClause in places.

whitehawk replied:
Ok, will swap

KnightMurloc commented:
does the added test run during regression tests? I can't find it in the logs.

whitehawk committed:
Simplify pg_stat_statements test

Removed redundant checks in pg_stat_statements test.
Plus minor refactoring in JumbleExpr.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61432

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/950363

whitehawk committed:
Add pg_stat_statements test to regression

whitehawk commented:
> does the added test run during regression tests? I can't find it in the logs.

Indeed, looks like the test is not executed automatically. Will fix it.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61438

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/950782

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/950783

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,132 @@
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item1, item2;
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item2, item3;
```
it seems the simple group by is not affected by the problem. therefore, I think this test can also be removed.

whitehawk committed:
Simplify pg_stat_statements test

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61485

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/953352

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/953353

whitehawk committed:
Simplify and refactor pg_stat_statements test

whitehawk committed:
Fix minor style issue in pg_stat_statements test

BenderArenadata commented:
Failed job Deploy multiarch Dockerimages: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/955119

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/955129

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/955130

whitehawk committed:
Fix jumble calculation for additional node tags

Problem description:
Jumbling logic doesn't handle properly queries with
 1) group_id() function call;
 2) calls of functions with 'anytable' parameter.
First type of queries causes warnings.
Second type of query causes error message "unrecognized RTE kind: 7".

Cause:
For first type of queries - not handled node tag T_GroupId in JumbleExpr.
For second type of queries - not handled range table type RTE_TABLEFUNCTION
in JumbleRangeTable and node tag T_TableValueExpr in JumbleExpr.

Fix:
For T_GroupId - added this tag to switch case inside JumbleExpr to
suppress warning. The node tag for it was already handled. Struct GroupId
doesn't have any additional fields, that could be added to jumble.
For RTE_TABLEFUNCTION - added 'functions' field of RangeTblEntry to the jumble.
RangeTblEntry in case of RTE_TABLEFUNCTION uses 'functions' and 'subquery'
fields (other signigicant fields are null). But 'subquery' is duplicated in
TableValueExpr below, so do not jumble it.
For T_TableValueExpr - added 'subquery' field of TableValueExpr (the subquery
that is inside "TABLE(<subquery>)" statement) to the jumble.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61797

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/967112

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/967110

KnightMurloc commented on contrib/pg_stat_statements/output/test_pg_stat_statements.source:
```diff
@@ -0,0 +1,139 @@
+(0 rows)
+
+-- check anytable parameter for a function
+CREATE FUNCTION multiset_1(a anytable) RETURNS TABLE(a int, b text)
```
Try using anytable_out as a multiset function. Its call always fails, so you need to make a query so that it is not called but is in the query.
Something like that. Is it possible to reproduce the problem with a similar query?
```sql
select count(1) from anytable_out( table(select 1) ) where 1 = 0;
```

whitehawk replied:
Indeed, it can help to simplify the test. Updated it.

whitehawk committed:
Simplify pg_stat_statements test

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/62084

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/979606

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/979608

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,58 @@
+--- check anytable parameter for a function
+SELECT pg_stat_statements_reset();
+
+SELECT COUNT(1) FROM anytable_out(TABLE(SELECT * FROM t)) WHERE 1 = 0;
+SELECT COUNT(1) FROM anytable_out(TABLE(SELECT * FROM t WHERE a=0)) WHERE 1 = 0;
```
I think `count` can be omitted and left just `*`. Also add a comment that the function cannot be called and that why we have always a false condition.

whitehawk replied:
done

whitehawk committed:
Improve pg_stat_statements test

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/62087

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/979746

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/979747

RekGRpth commented on src/backend/utils/misc/queryjumble.c:
```diff
@@ -597,6 +598,36 @@ JumbleExpr(JumbleState *jstate, Node *node)
+			 */
+			break;
+		case T_Integer:
+			{
```
The braces are unnecessary here because no variables are created on the stack.

whitehawk replied:
fixed

RekGRpth commented on contrib/pg_stat_statements/expected/test_pg_stat_statements.out:
```diff
@@ -0,0 +1,131 @@
+SELECT query, calls FROM pg_stat_statements ORDER BY query;
+                       query                       | calls 
+---------------------------------------------------+-------
+ SELECT GROUPING (a) FROM t GROUP BY ROLLUP(a, b); |     1
```
These three lines are tested below, so I suggest removing them from here (or add reset after this check).

whitehawk replied:
added reset

RekGRpth commented on contrib/pg_stat_statements/expected/test_pg_stat_statements.out:
```diff
@@ -0,0 +1,131 @@
+(3 rows)
+
+-- check group_id() in query
+SELECT group_id() AS c FROM t GROUP BY GROUPING SETS ((a), (a,b));
```
What does this request check if a reset is made after it?

whitehawk replied:
It checks that no warning appear if group_id() function is called (before my change this query was executed with warning).

RekGRpth replied:
why not include it to some of `SELECT query, calls FROM pg_stat_statements ORDER BY query;`?

whitehawk replied:
Ok, moved it

whitehawk committed:
Improve pg_stat_statements test + minor code style fix

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/62194

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/984402

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/984403

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/984396

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/986170

KnightMurloc commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,61 @@
+-- check group_id() in a query
+SELECT group_id() FROM t GROUP BY ROLLUP(a, b);
+
+-- check that 2 queries have separate entries
```
```suggestion
-- check that 3 queries have separate entries
```

whitehawk committed:
Fix comment in pg_stat_statements test

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/62293

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/988007

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/988008

RekGRpth commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,61 @@
+--- check anytable parameter for a function
+SELECT pg_stat_statements_reset();
+
+-- call of anytable_out will cause an error,
```
What is the practical purpose of this test and the fix for it?

whitehawk replied:
It checks  'anytable' parameter (SELECT statement enclosed in a TABLE() function). 
Fix for it - adding RTE_TABLEFUNCTION and T_TableValueExpr handling.

Commit:
Merge branch 'adb-6.x-dev' into ADBDEV-4486

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/62408

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/992651

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/992648

andr-sokolov commented on contrib/pg_stat_statements/sql/test_pg_stat_statements.sql:
```diff
@@ -0,0 +1,61 @@
+
+-- Known issue: query is not added to pg_stat_statements statistics in
+-- case it is planned by GPORCA. So disable GPORCA during tests.
+SET optimizer='off';
```
Remove please the quotes, because the optimizer GUC is set without them in all other tests

whitehawk committed:
Fix minor code style issue in pg_stat_statements test

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/62478

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/996082

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/996083

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/996076

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/996080
